skimage/morphology/binary.py

```diff
@@ -38,24 +36,13 @@ def binary_erosion(image, selem=None, out=None):
         The result of the morphological erosion with values in ``[0, 1]``.
 
     """
-
-    selem = (selem != 0)
-    selem_sum = np.sum(selem)
-
-    if selem_sum <= 255:
-        conv = np.empty_like(image, dtype=np.uint8)
-    else:
-        conv = np.empty_like(image, dtype=np.uint)
```

Comment:
@stefanv there won't be any overflow because we don't use `convolve` anymore, but rather `binary_erosion` directly. There's a test for overflow and it passes.